packages/components/tag/src/tag-body.tsx

```diff
@@ -1,16 +1,40 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import React, {
+  MouseEvent,
+  KeyboardEvent,
+  ReactNode,
+  ElementType,
+} from 'react';
 import styled from '@emotion/styled';
-import { css, useTheme } from '@emotion/react';
+import type { Theme } from '@emotion/react';
+import type { LocationDescriptor } from 'history';
+import { css, useTheme, SerializedStyles } from '@emotion/react';
 import {
   customProperties as vars,
   designTokens,
 } from '@commercetools-uikit/design-system';
 import Text from '@commercetools-uikit/text';
 
-const Body = styled.div``;
+type TTagBodyProps = {
+  as?: ElementType;
+  to?: string | LocationDescriptor;
+  type: string;
+  onClick?: (
+    event: MouseEvent<HTMLDivElement> | KeyboardEvent<HTMLDivElement>
+  ) => void;
+  onRemove: (
+    event: MouseEvent<HTMLDivElement> | KeyboardEvent<HTMLDivElement>
+  ) => void;
+  isDisabled: boolean;
+  children: ReactNode;
+  styles?: Record<string, SerializedStyles>;
+};
+
+const Body = styled.button``;
```

Comment:
Here I have tried with various other elements like `styled.a` `styled.link`. But they are also having the same problem.

Reply:
What were you trying to achieve?

Reply:
We are trying to render the component as `react-router-dom`'s `Link` if there is an url as `to` property and it's not disabled. We are also having `onClick` property.

The problem is `Property 'to' does not exist on any element`. I have gone through types rabbit hole and found out that none of the `HTMLAnchorElement`, `HTMLButtonElement`, etc has a key called `to`. So those are rejecting this prop.